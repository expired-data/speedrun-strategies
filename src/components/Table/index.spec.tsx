import React from "react";
import { render } from "@testing-library/react";
import { Table } from "./";
import { Column } from "react-table";

test("Table renders without data", () => {
  expect(
    render(<Table data={[]} columns={[]} />).container.innerHTML
  ).toMatchSnapshot();
});

test("Table renders with sample data", () => {
  const data = [
    {
      name: "foo",
      age: 12,
    },
  ];

  const columns: Column<{ name: string; age: number }>[] = [
    { accessor: "name", Header: "Name" },
    { accessor: "age", Header: "Age" },
  ];

  expect(
    render(<Table data={data} columns={columns} />).container.innerHTML
  ).toMatchSnapshot();
});
