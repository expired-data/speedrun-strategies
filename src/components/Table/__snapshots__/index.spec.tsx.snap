// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders with sample data 1`] = `"<table class=\\"sc-bdfBwQ iYunag\\" role=\\"table\\"><thead><tr role=\\"row\\" class=\\"sc-gsTCUz kGUbwO\\"><th colspan=\\"1\\" role=\\"columnheader\\">Name</th><th colspan=\\"1\\" role=\\"columnheader\\">Age</th></tr></thead><tbody role=\\"rowgroup\\"><tr role=\\"row\\" class=\\"sc-gsTCUz dGdWN\\"><td role=\\"cell\\" class=\\"sc-dlfnbm hWxGDI\\">foo</td><td role=\\"cell\\" class=\\"sc-dlfnbm hWxGDI\\">12</td></tr></tbody></table>"`;

exports[`Table renders without data 1`] = `"<table class=\\"sc-bdfBwQ iYunag\\" role=\\"table\\"><thead></thead><tbody role=\\"rowgroup\\"></tbody></table>"`;
